### PR TITLE
Change TLS dir to same as Qt config

### PR DESCRIPTION
**Pull request by nbolton**
OMG... how many places is this dir string defined? Will fix that in a separate PR.

Fixes: #5762

### CODE REVIEW

**Comment by sithlord48** [comment]
I fixed this in [inputleap](https://github.com/input-leap/input-leap/pull/1892) there was alot of extra defines and if def guards on the name when they even were the same

File: src/lib/arch/unix/ArchFileUnix.cpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ std::string ArchFileUnix::getProfileDirectory() {
     dir = m_profileDirectory;
   } else {
 #if WINAPI_XWINDOWS
-    dir = getUserDirectory().append("/." DESKFLOW_APP_ID);
+    dir = getUserDirectory().append("/.config/" DESKFLOW_APP_NAME);
```
Comment (sithlord48) [inline src/lib/arch/unix/ArchFileUnix.cpp:108]: Use `$XDG_CONFIG_DIR` if defined or `~/.config` if its not

**Review by sithlord48 — CHANGES_REQUESTED** [review]

File: src/lib/arch/unix/ArchFileUnix.cpp (hunk 1 above)
Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp:108]: How about `$XDG_CONFIG_DIR` or fall back to `~/.config/Deskflow`?

Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp:108]: I'll rip it from your IL PR.

Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp]: Done: bcc41ea

**Review by sithlord48 — COMMENTED** [review]
Looks good did not run it

**Comment by nbolton** [comment]
> Looks good did not run it

Tested on Linux with client and server, also tested with XDG_CONFIG_HOME set and seems to work well.

File: src/lib/arch/unix/ArchFileUnix.cpp [hunk 2]
```diff
@@ -100,17 +101,21 @@ std::string ArchFileUnix::getPluginDirectory() {
...
-    dir = getUserDirectory().append("/." DESKFLOW_APP_ID);
+    const auto xdgDir = std::getenv("XDG_CONFIG_HOME");
+    if (xdgDir != nullptr) {
+      return std::filesystem::path(xdgDir) / DESKFLOW_APP_NAME;
+    } else {
+      return homeDir / ".config" / DESKFLOW_APP_NAME;
+    }
 #else
-    dir = getUserDirectory().append("/Library/" DESKFLOW_APP_NAME);
+    return homeDir / "Library" / DESKFLOW_APP_NAME;
```
Comment (sithlord48) [inline src/lib/arch/unix/ArchFileUnix.cpp:116]: This needs to be double checked iirc its `~/Library/Preferences` for user and /Library/Preferences` for global settings on mac os

Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp:116]: Ah, I belive the `~/Library/Preferences` dir is just for `.plist` files.

Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp:116]: Seems that `~/Library/Application Support` is an alternative, but seems like plenty of apps just make a dir in root of `~/Library`, so looks like common practice. Not sure what Apple wants us to do though...

Comment (sithlord48) [inline src/lib/arch/unix/ArchFileUnix.cpp:116]: Shouldn't this be something like 
return QStringLitera("%1/Library/%2).arg(homeDir, DESKFLOW_APP_NAME);

**Review by sithlord48 — CHANGES_REQUESTED** [review]

File: src/lib/arch/unix/ArchFileUnix.cpp (hunk 2 above)
Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp:116]: If it was in Qt, sure, but that's not the case.

Comment (sithlord48) [reply src/lib/arch/unix/ArchFileUnix.cpp:116]: append says otherwise

Comment (sithlord48) [reply src/lib/arch/unix/ArchFileUnix.cpp:116]: well if its not should there some concat or something?

Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp:116]: I'm confused, you mean [`std::string:append`](https://cplusplus.com/reference/string/string/append/)?

**Review by sithlord48 — APPROVED** [review]

File: src/lib/arch/unix/ArchFileUnix.cpp (hunk 2 above)
Comment (nbolton) [reply src/lib/arch/unix/ArchFileUnix.cpp:116]: Implicit string conversion: https://en.cppreference.com/w/cpp/filesystem/path/string

**Pull request merged**